cmd/soroban-cli/src/commands/contract/deploy/wasm.rs

```diff
@@ -115,18 +119,20 @@ impl NetworkRunnable for Cmd {
         &self,
         global_args: Option<&global::Args>,
         config: Option<&config::Args>,
-    ) -> Result<String, Error> {
+    ) -> Result<TxnResult<String>, Error> {
         let config = config.unwrap_or(&self.config);
         let wasm_hash = if let Some(wasm) = &self.wasm {
+            let mut fee = self.fee.clone();
+            fee.build_only = false;
```

Comment:
This results in a tx being sent to the network, to do the install, when --build-only is passed which would be surprising for someone wanting to only produce a tx. When building only we should probably skip install and build just the deployment transaction. Then they can produce the install and deploy txs independently with the two commands. Wdyt?

Reply:
Yeah it would be nice to have these as two operations in the same transaction, but I agree I've disliked the install step because it is an unexpected side effect.

Reply:
I think we can skip the install when deploy is build only. We don't actually need anything from the install. The hash is all the deploy needs and it can be calculated with a sha256 of the binary.